queue/src/test/java/org/killbill/bus/TestRetries.java

```diff
@@ -0,0 +1,193 @@
+                           Iterators.size(notificationSqlDao.getHistoricalQueueEntriesForSearchKeys("notifications-retries:myEvent-listener", SEARCH_KEY_1, SEARCH_KEY_2, notificationQueueConfig.getHistoryTableName())) == 3;
+                }
+            });
+
+            // Initial event was processed once
+            List<BusEventModelDao> historicalEntriesForOriginalEvent = ImmutableList.<BusEventModelDao>copyOf(busSqlDao.getHistoricalQueueEntriesForSearchKeys(SEARCH_KEY_1, SEARCH_KEY_2, persistentBusConfig.getHistoryTableName()));
+            Assert.assertEquals(historicalEntriesForOriginalEvent.size(), 1);
+            Assert.assertEquals((long) historicalEntriesForOriginalEvent.get(0).getErrorCount(), (long) 0);
+            // State is initially FAILED
+            Assert.assertEquals(historicalEntriesForOriginalEvent.get(0).getProcessingState(), PersistentQueueEntryLifecycleState.FAILED);
+
+            // Retry events
+            List<NotificationEventModelDao> historicalEntriesForRetries = ImmutableList.<NotificationEventModelDao>copyOf(notificationSqlDao.getHistoricalQueueEntriesForSearchKeys("notifications-retries:myEvent-listener", SEARCH_KEY_1, SEARCH_KEY_2, notificationQueueConfig.getHistoryTableName()));
+            Assert.assertEquals(historicalEntriesForRetries.size(), 3);
+            for (final NotificationEventModelDao historicalEntriesForRetry : historicalEntriesForRetries) {
+                Assert.assertEquals((long) historicalEntriesForRetry.getErrorCount(), (long) 0);
+                Assert.assertEquals(historicalEntriesForRetry.getProcessingState(), PersistentQueueEntryLifecycleState.FAILED);
+            }
+
+            // Make the next retry work
+            retryableBusService.shouldFail(false);
+
+            clock.addDays(1);
+
+            // Make sure all notifications are processed
+            await().atMost(10, TimeUnit.SECONDS).until(new Callable<Boolean>() {
+                @Override
+                public Boolean call() throws Exception {
+                    return Iterators.size(busSqlDao.getHistoricalQueueEntriesForSearchKeys(SEARCH_KEY_1, SEARCH_KEY_2, persistentBusConfig.getHistoryTableName())) == 1 &&
+                           Iterators.size(notificationSqlDao.getHistoricalQueueEntriesForSearchKeys("notifications-retries:myEvent-listener", SEARCH_KEY_1, SEARCH_KEY_2, notificationQueueConfig.getHistoryTableName())) == 4;
+                }
+            });
+
+            // Initial event was processed once
+            historicalEntriesForOriginalEvent = ImmutableList.<BusEventModelDao>copyOf(busSqlDao.getHistoricalQueueEntriesForSearchKeys(SEARCH_KEY_1, SEARCH_KEY_2, persistentBusConfig.getHistoryTableName()));
+            Assert.assertEquals(historicalEntriesForOriginalEvent.size(), 1);
+            Assert.assertEquals((long) historicalEntriesForOriginalEvent.get(0).getErrorCount(), (long) 0);
+            // State is still FAILED
```

Comment:
Ideally it should be `PROCESSED` , but i remember this is what you explained and good enough probably